frontend/src/features/admin-form/responses/ResponsesPage.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useState } from 'react'
+import { useParams } from 'react-router'
+import { Skeleton } from '@chakra-ui/react'
+import { wrap } from 'comlink'
+// Need to use inline loading in order to use comlink syntax
+// Typescript documentation for worker-loader webpack v4 also seems outdated,
+// loading the worker script via webpack config no longer works
+// eslint-disable-next-line import/no-webpack-loader-syntax
+import DecryptionWorker from 'worker-loader!./workers/decryption.worker'
+
+import { StorageModeSubmissionMetadata } from '~shared/types/submission'
+
+import Button from '~components/Button'
+
+import { useAdminFormSettings } from '../settings/queries'
+
+import { WorkerInterface } from './workers/type/workerinterface'
+import { killWorkers } from './AdminSubmissionsService'
+//import { downloadEncryptedResponses } from './AdminSubmissionsService'
+import { useFormResponses } from './queries'
+
+const useDecryptionWorkers = () => {
+  const [workers, setWorkers] = useState<WorkerInterface[]>([])
+
+  const downloadEncryptedResponses = (
+    formId: string,
+    formTitle: string,
+    secretKey: string,
+  ) => {
+    if (workers.length) killWorkers(workers)
+
+    const numWorkers = window.navigator.hardwareConcurrency || 4
+
+    const workerPool: WorkerInterface[] = []
+
+    // Workerpool sample setup
+    for (let i = 0; i < numWorkers; i++) {
+      const worker = new DecryptionWorker()
+      const workerApi =
+        wrap<import('./workers/decryption.worker').DecryptionWorker>(worker)
+      workerPool.push({ worker, workerApi })
```

Comment:
Is there a point to also adding the `worker` constant into the map? Feels unnecessary and never gets used at all.
I think it would be cleaner if we just have an array of workers?

suggest
```ts
const worker = wrap<DecryptionWorker>(new DecryptionWorker())
workerPool.push(worker)
```

Reply:
The worker constant needs to be kept in order to call `terminate` on the workers

Reply:
I see, missed that part. Can't we then just store worker in workerPool and call wrap on the worker when we actually access the functions?

Reply:
I think that will work!